tests/AutoReview/FixerTest.php

```diff
@@ -74,23 +97,78 @@ public function testFixerDefinitions(FixerInterface $fixer)
                 $sample instanceof FileSpecificCodeSampleInterface ? $sample->getSplFileInfo() : $dummyFileInfo,
                 $tokens
             );
-            $this->assertTrue($tokens->isChanged(), sprintf('[%s] Sample #%d is not changed during fixing.', $fixer->getName(), $sampleCounter));
+            $this->assertTrue($tokens->isChanged(), sprintf('[%s] Sample #%d is not changed during fixing.', $fixerName, $sampleCounter));
 
             $duplicatedCodeSample = array_search(
                 $sample,
                 array_slice($samples, 0, $sampleCounter),
                 false
             );
+
             $this->assertFalse(
                 $duplicatedCodeSample,
-                sprintf('[%s] Code sample #%d duplicates #%d.', $fixer->getName(), $sampleCounter, $duplicatedCodeSample)
+                sprintf('[%s] Code sample #%d duplicates #%d.', $fixerName, $sampleCounter, $duplicatedCodeSample)
             );
         }
 
+        if ($fixerIsConfigurable) {
+            $options = $fixer->getConfigurationDefinition()->getOptions();
+
+            if (isset($configSamplesProvided['default'])) {
+                reset($configSamplesProvided);
+                $firstKey = key($configSamplesProvided);
+
+                $this->assertArrayHasKey('default', $configSamplesProvided, sprintf('[%s] Configurable fixer has no sample with default configuration.', $fixerName));
+                $this->assertSame('default', $firstKey, 'First code sample must be for the default configuration.');
+            } else {
+                $allOptionsHaveDefault = true;
+                /** @var FixerOption $option */
+                foreach ($options as $option) {
+                    if (!$option->hasDefault()) {
+                        $allOptionsHaveDefault = false;
+
+                        break;
+                    }
+                }
+
+                if ($allOptionsHaveDefault) {
+                    $this->fail(sprintf('[%s] Has no sample for default configuration but all options have a default setting.', $fixerName));
+                }
+            }
+
+            // test that all configuration options are used at least one time in the samples
+
+            $optionsNames = array();
+            /** @var FixerOption $option */
+            foreach ($options as $option) {
+                $optionsNames[$option->getName()] = $option->hasDefault() ? 1 : 0;
+            }
+
+            if (count($optionsNames)) {
```

Comment:
don't think so, `options` will always be set here (with a count), however we want to know if after filtering there are still options left